components/places/src/storage/history_metadata.rs

```diff
@@ -4,14 +4,82 @@
 
 use crate::db::{PlacesDb, PlacesTransaction};
 use crate::error::Result;
-use crate::msg_types::{HistoryMetadata, HistoryMetadataList, HistoryMetadataObservation};
-use sql_support::{self, ConnExt};
+use rusqlite::types::{FromSql, FromSqlResult, ToSql, ToSqlOutput, ValueRef};
+use sql_support::ConnExt;
+use std::vec::Vec;
 use sync_guid::Guid as SyncGuid;
 use types::Timestamp;
 use url::Url;
 
 use lazy_static::lazy_static;
 
+#[derive(Copy, Clone, Debug, PartialEq)]
+pub enum DocumentType {
+    Regular = 0,
+    Media = 1,
+}
+
+impl FromSql for DocumentType {
+    #[inline]
+    fn column_result(value: ValueRef<'_>) -> FromSqlResult<Self> {
+        Ok(match value.as_i64()? {
+            0 => DocumentType::Regular,
+            1 => DocumentType::Media,
+            other => {
+                // seems safe to ignore?
```

Comment:
Stepping maybe too far ahead, I guess that can happen if other clients implement document types we do not, in which case we _may_ rewrite them to `regular` as a result of this conversion - depending on how these results are used, although not likely currently (e.g. these are never written back by the api consumers). And we don't want that kind of misalignment to happen at all, really, so perhaps this is moot.

Reply:
FTR, I think ignoring it is the wrong approach - it's not what other enums in this crate do, so there's no good reason to set a new precedent. Getting unknown values in via sync is an interesting issue, but one Sync should explicitly deal with rather than writing "invalid" enum values. We could even consider using a `CHECK` constraint in the sql?